src/error.rs

```diff
@@ -71,6 +82,8 @@ pub enum Error {
     /// A standard error with a string description;
     /// a more specific error should generally be used.
     DefaultError(String),
+    #[cfg(feature = "ssl")]
+    SslError(String),
```

Comment:
Do you you have an example of an OpenSSL error in default formatting vs. the one implemented here? Just wondering if it might be preferable to remove the special case and have these errors be coerced to `io::Error` automatically.